linera-core/src/client/mod.rs

```diff
@@ -380,16 +380,12 @@ where
             let mut result = self.handle_certificate(certificate.clone(), vec![]).await;
 
             result = match &result {
-                Err(err) => {
-                    if let Some(blob_ids) = err.get_blobs_not_found() {
-                        let blobs = remote_node.try_download_blobs(blob_ids.as_slice()).await;
-                        if blobs.len() != blob_ids.len() {
-                            result
-                        } else {
-                            self.handle_certificate(certificate, blobs).await
-                        }
-                    } else {
+                Err(LocalNodeError::BlobsNotFound(blob_ids)) => {
+                    let blobs = remote_node.try_download_blobs(blob_ids.as_slice()).await;
+                    if blobs.len() != blob_ids.len() {
```

Comment:
this is weird but changed in the next PR, right?

Reply:
I think @afck had a PR that fixed this behavior, so this is already gone in my local version of this PR. Will update it in a bit!

Reply:
Didn't I already merge that? I think `try_download_blobs` returns `None` if it couldn't download all requested blobs? So I think you can remove this comparison.

Reply:
I think that's what I said 😅 already have that done locally, but haven't updated the PR yet